server.js

```diff
@@ -144,12 +147,77 @@ storage.initSync();
 
 var _ = require('lodash');    // JavaScript utility library
 var dgram = require('dgram'); // UDP Broadcasting library
-var ips = {activeInterface: 'en0', interfaces: {}};
-if (storage.getItemSync('activeNetworkInterface') !== undefined) {
-    //console.log( storage.getItemSync('activeNetworkInterface'));
-    ips.activeInterface = storage.getItemSync('activeNetworkInterface');
+
+var services = {};
+if (!isMobile) {
+    services.networkInterface = require('network-interfaces');
+}
+services.options = {ipVersion: 4};
+services.ips = {activeInterface: null, tempActiveInterface: null, interfaces: {}};
+services.ip = null
+services.updateAllObjcts = function (ip){
+    console.log("updating all objects with new IP: ", ip);
+    for (let key in objects){
+        objects[key].ip = ip;
+    }
+};
+services.getIP = function (){
+    this.ips.interfaces = {};
+    // if this is mobile, only allow local interfaces
+    if (isMobile) {
+        this.ips.interfaces["mobile"] = '127.0.0.1';
+        this.ips.activeInterface = "mobile";
+        return '127.0.0.1';
+    }
+    
+    // Get All available interfaces
+        var interfaceNames = this.networkInterface.getInterfaces(this.options);
```

Comment:
options can be a local variable instead of stored on this object

Reply:
sure